src/identity/storage/WebIdAdapterFactory.ts

```diff
@@ -85,24 +83,10 @@ export class WebIdAdapter implements Adapter {
   }
 
   private async parseRdfWebId(data: string, id: string, response: Response): Promise<AdapterPayload> {
-    const contentType = response.headers.get('content-type');
-    if (!contentType) {
-      throw new Error(`No content-type received for client WebID ${id}`);
-    }
-
-    // Try to convert to quads
-    const representation = new BasicRepresentation(data, contentType);
-    const preferences = { type: { [INTERNAL_QUADS]: 1 }};
-    const converted = await this.converter.handleSafe({ representation, identifier: { path: id }, preferences });
-    const quads = new Store();
-    const importer = quads.import(converted.data);
-    await new Promise((resolve, reject): void => {
-      importer.on('end', resolve);
-      importer.on('error', reject);
-    });
+    const store = await fetchDataset(response, this.converter, data);
 
     // Find the valid redirect uris
-    const match = quads.getObjects(id, 'http://www.w3.org/ns/solid/oidc#redirect_uris', null);
+    const match = store.getObjects(id, 'http://www.w3.org/ns/solid/oidc#redirect_uris', null);
```

Comment:
Move to constants?

Reply:
I would keep it in this class (for now at least) since it's very specific to the behaviour that is only covered in this class (that is the handling of client_id WebIDs).